src/java/org/apache/cassandra/repair/StreamingRepairTask.java

```diff
@@ -88,7 +88,7 @@ StreamPlan createStreamPlan(InetAddressAndPort dest)
         if (!asymmetric)
             // see comment on RangesAtEndpoint.toDummyList for why we synthesize replicas here
             sp.transferRanges(dest, desc.keyspace, RangesAtEndpoint.toDummyList(ranges), desc.columnFamily); // send ranges to the remote node
-        return sp;
+        return sp.withPlanId(desc.determanisticId());
```

Comment:
the idea is that streaming done by repair will use the RepairJob id (as defined by CASSANDRA-15399), so if a repair exists with the same planId, then this stream can map to the repair without any networking changes